File: C/Question Practice C/Frog_Jump.c
#include<stdio.h>
#include<stdlib.h>

void FrogJump(int arr[], int solutions[], int size, int index, int energy, int* counter) {
    if (index >= size-1) {
        solutions[(*counter)++] = energy;
        return;
    }

    // Jump to the next stone
    if (index + 1 < size) FrogJump(arr, solutions, size, index + 1, energy + abs(arr[index] - arr[index + 1]), counter);

    // Jump to the stone after next
    if (index + 2 < size) FrogJump(arr, solutions, size, index + 2, energy + abs(arr[index] - arr[index + 2]), counter);
}

int main(){
    int size, counter = 0;
    printf("Enter the no of Elements: ");
    scanf("%d", &size);

    int arr[size], solutions[size*size]; // Maximum possible solutions

    for (int i = 0; i < size; i++) {
        printf("Enter the Element: ");
        scanf("%d", &arr[i]);
    }

    printf("\nThe Elements in the array are:\n");
    for (int i = 0; i < size; i++) printf("%d ", arr[i]);

    FrogJump(arr, solutions, size, 0, 0, &counter);
    int minEnergy = solutions[0];
    for (int i = 1; i < counter; i++) if (solutions[i] < minEnergy) minEnergy = solutions[i];
    printf("\n\nMinimum Energy Required: %d", minEnergy);

    return 0;
}

File: Java/Question Practice Java/Frog_Jump.java
import java.util.Scanner;
public class Frog_Jump{
    public static void FrogJump(int[] arr, int[] solutions, int[] counter, int index, int energy) {
        if (index >= arr.length - 1) {
            solutions[counter[0]++] = energy;
            return;
        }

        // Jump 1 step
        FrogJump(arr, solutions, counter, index + 1, energy + Math.abs(arr[index] - arr[index + 1]));

        // Jump 2 steps
        if (index + 2 < arr.length) FrogJump(arr, solutions, counter, index + 2, energy + Math.abs(arr[index] - arr[index + 2]));
    }

    public static void main(String[] args){
        Scanner input = new Scanner(System.in);

        System.out.print("Enter the no of Elements: ");
        int size = input.nextInt();

        int[] arr = new int[size];
        int[] solutions = new int[size*size];
        int[] counter = new int[1];
        counter[0] = 0;
        for (int i = 0; i < size; i++) {
            System.out.print("Enter the Element: ");
            arr[i] = input.nextInt();
        }

        System.out.println("\nElements in the array are:");
        for (int val : arr) System.out.print(val + " ");

        FrogJump(arr, solutions, counter, 0, 0);
        int minEnergy = solutions[0];
        for (int i = 1; i < counter[0]; i++) if (solutions[i] < minEnergy) minEnergy = solutions[i];
        System.out.println("\n\nMinimum Energy Required: " + minEnergy);
    }
}